sciplot/Plot.hpp

```diff
@@ -50,40 +50,28 @@
 #include <sciplot/specs/TicsSpecsMajor.hpp>
 #include <sciplot/specs/TicsSpecsMinor.hpp>
 #include <sciplot/Utils.hpp>
+#include <sciplot/PlotBase.hpp>
 
 namespace sciplot {
 
 /// The class used to create a plot containing graphical elements.
-class Plot
+class Plot : public PlotBase
 {
   public:
     /// Construct a default Plot object
     Plot();
 
-    /// Set the palette of colors for the plot.
-    /// @param name Any palette name displayed in https://github.com/Gnuplotting/gnuplot-palettes, such as "viridis", "parula", "jet".
-    auto palette(std::string name) -> void;
-
-    /// Set the size of the plot (in unit of points, with 1 inch = 72 points).
-    auto size(std::size_t width, std::size_t height) -> void;
-
-    /// Set the font name for the plot (e.g., Helvetica, Georgia, Times).
-    auto fontName(std::string name) -> void;
-
-    /// Set the font size for the plot (e.g., 10, 12, 16).
-    auto fontSize(std::size_t size) -> void;
-
     /// Set the label of the x-axis and return a reference to the corresponding specs object.
-    auto xlabel(std::string label) -> AxisLabelSpecs&;
+    auto xlabel(const std::string &label) -> AxisLabelSpecs&;
```

Comment:
I tend not to use `const&` for lightweight data types (we won't have label strings, filename strings, etc., with megabytes of data!). This keeps the list of function arguments simpler.

Reply:
Sorry if that annoys you, it's just a habit with me.
I do a lot of performance critical stuff. Apart from that I wouldn't say that std::string is a lightweight datatype. Every copy operation requires a memory allocation on the heap, which is a slow operation in itself, regardless of the size of the allocation. In general, you should avoid many small memory allocations, because they can cause memory fragmentation.  (Yes, I know, there must be a lot of them, but I have often reached that limit).

But we do it the way you like it.

Reply:
I use "const &" for everything, except for basic types like char, int, float. I would in 99% of the cases pass std::string by reference.

Reply:
Alright, we go with `const std::string&` to be on the safe side.